app/hooks/use-listen-ledger.ts

```diff
@@ -0,0 +1,8 @@
+import { useEffect } from 'react';
+
+export function useListenLedger() {
```

Comment:
small nit, I personally prefer to name any hook that is only effect with the suffix "Effect", so for this one, `useListenLedgerEffect` or `useLedgerEffect` reads more clear to me.